From 44172505db14959b9848a3eacc8e4c0b79e61249 Mon Sep 17 00:00:00 2001
From: Hebbar, Gururaja <gururaja.hebbar@ti.com>
Date: Mon, 3 Oct 2011 18:20:27 +0530
Subject: [PATCH 27/29] ARM:omap:am335x: MMC: enable proper WP detection

Since WP Detection is happening properly, enable support to detect
correct detection

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 drivers/mmc/host/omap_hsmmc.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/mmc/host/omap_hsmmc.c b/drivers/mmc/host/omap_hsmmc.c
index 94b30a8..c8a95f3 100644
--- a/drivers/mmc/host/omap_hsmmc.c
+++ b/drivers/mmc/host/omap_hsmmc.c
@@ -250,14 +250,10 @@ static int omap_hsmmc_get_wp(struct device *dev, int slot)
 		/* NOTE: assumes write protect signal is active-high */
 		return gpio_get_value_cansleep(mmc->slots[0].gpio_wp);
 	else {
-#if 0
 		pstate = 0;
 		pstate = OMAP_HSMMC_READ(host->base, PSTATE);
 		pstate &= PSTATE_WP_MASK;
 		return !(pstate >> PSTATE_WP_SHIFT);
-#endif
-		udelay(100);
-		return 0;
 	}
 }
 
-- 
1.6.6.1

